server/api/region.go

```diff
@@ -257,6 +258,61 @@ func (h *regionsHandler) GetEmptyRegion(w http.ResponseWriter, r *http.Request)
 	h.rd.JSON(w, http.StatusOK, regionsInfo)
 }
 
+type regionSize struct {
+	Size  int64 `json:"size"`
+	Count int   `json:"count"`
+}
+
+type regionKeys struct {
+	Keys  int64 `json:"keys"`
+	Count int   `json:"count"`
+}
+
+type regionHist struct {
+	SizeHist []*regionSize `json:"region size histogram:"`
+	KeysHist []*regionKeys `json:"region keys histogram:"`
+}
+
+func (h *regionsHandler) GetHistogram(w http.ResponseWriter, r *http.Request) {
+	rc := getCluster(r.Context())
+	regions := rc.GetRegions()
+	sizeHistgoram := histogram.New(10)
```

Comment:
How the bounds of bins is determined?

Reply:
The bins boundary code is in the method compress () of github.com/beorn7/perks/histogram

```
minGapIndex := -1
minGap := math.MaxFloat64
for i := 0; i < r.bins.Len()-1; i++ {
   gap := gapWeight(r.bins[i], r.bins[i+1])
   if minGap > gap {
      minGap = gap
      minGapIndex = i
   }
}
prev := r.bins[minGapIndex]
next := r.bins.remove(minGapIndex + 1)
prev.Update(next)
```

Its boundaries are dynamic. Every time new data comes in, it will find the smallest boundary value of the two adjacent bins, then delete the next bin, and insert the value of this bin into the previous bin.